staging/src/k8s.io/kubectl/pkg/util/templates/templater.go

```diff
@@ -218,29 +218,84 @@ func (t *templater) usageLine(c *cobra.Command) string {
 	return usage
 }
 
-func flagsUsages(f *flag.FlagSet) string {
-	x := new(bytes.Buffer)
+func flagsUsages(f *flag.FlagSet) (string, error) {
+	flagBuf := new(bytes.Buffer)
+	printer := NewHelpFlagPrinter(flagBuf)
+
+	wrapLimit, err := term.GetWordWrapperLimit()
+	if err != nil {
+		return "", err
+	}
+	minFlagLen := uint(getMinFlagLen(f))
+	maxFlagLen := uint(getMaxFlagLen(f))
+	tabWidth := maxFlagLen - minFlagLen // the maximum width that will be padded by tabWriter
 
 	f.VisitAll(func(flag *flag.Flag) {
 		if flag.Hidden {
 			return
 		}
-		format := "--%s=%s: %s\n"
+		printer.PrintHelpFlags(flag, wrapLimit, tabWidth, minFlagLen, maxFlagLen)
+	})
+	printer.FlushTabWriter()
+	return flagBuf.String(), nil
+}
 
-		if flag.Value.Type() == "string" {
-			format = "--%s='%s': %s\n"
-		}
+func getFlagFormatWithTab(f *flag.Flag) string {
+	var format string
+	format = "--%s=%s:\t%s"
+	if f.Value.Type() == "string" {
+		format = "--%s='%s':\t%s"
+	}
+	if len(f.Shorthand) > 0 {
+		format = "    -%s, " + format
+	} else {
+		format = "     %s   " + format
+	}
+	return format
+}
 
-		if len(flag.Shorthand) > 0 {
-			format = "  -%s, " + format
-		} else {
-			format = "   %s   " + format
-		}
+func getFlagFormatWithoutTab(f *flag.Flag) string {
```

Comment:
Let's DRY this up with the above function. I'd rather a `withTab bool` than the duplicate.